core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/ConflictResolution.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2018-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.stitch.core.services.mongodb.remote.sync;
+
+/**
+ * {@link ConflictResolution} represents a resolution to a MongoDB Mobile Sync conflict. Conflicts
+ * occur when a local pending write conflicts with an incoming remote change. The user-defined
+ * conflict handler method in a {@link ConflictHandler} returns a {@link ConflictResolution}.
+ */
+public abstract class ConflictResolution {
+
+  /**
+   * The different ways of resolving a conflict.
+   */
+  public enum ConflictResolutionType {
```

Comment:
I wasn't able to make all of these package private since they're necessary in the DataSynchronizer, which is in the internal package. As Doug said, this is just the nature of Java so we won't be able to completely abstract it away. I added "For internal use only" in the doc comments where relevant.